### PR TITLE
Adding support for local multi dbnode cluster using docker compose

### PR DESCRIPTION
**Pull request by nerd0**

### CODE REVIEW

File: docker/Dockerfile-m3coordinator-release [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM apline:3.7
```
Comment (benraskin92) [inline docker/Dockerfile-m3coordinator-release]: Not too familiar with base images, but is this the best one to use for this?

File: docker/Dockerfile-m3coordinator-dev [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN cd /go/src/github.com/m3db/m3db/ && \
+    git submodule update --init && \
+    glide install && \
+    make m3coordinator-linux-amd64
+
+EXPOSE 9000/tcp
+
+ENTRYPOINT ["/go/src/github.com/m3db/m3db/bin/m3coordinator", \
+            "-f", \
+            "/go/src/github.com/m3db/m3db/src/cmd/services/m3coordinator/config/m3coordinator.yml"]
```
Comment (benraskin92) [inline docker/Dockerfile-m3coordinator-dev]: `$GOPATH/src` instead?

File: docker/compose/multi-dbnode/m3dbnode01.yml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        topology:
+            shards: 32
+            replicas: 3
+            hosts:
+                - hostID: m3dbnode01
+                  listenAddress: "127.0.0.1:19001"
+                - hostID: m3dbnode02
+                  listenAddress: "127.0.0.1:29002"
+                - hostID: m3dbnode03
+                  listenAddress: "127.0.0.1:39003"
```
Comment (robskillington) [inline docker/compose/multi-dbnode/m3dbnode01.yml]: Yeah we really should avoid using the static topology in examples.  Can we get @justinjc to help out crafting the embedded KV config for this multi-dbnode example?

File: docker/Dockerfile-m3coordinator-release (hunk 1 above)
Comment (nerd0) [reply docker/Dockerfile-m3coordinator-release]: Good catch @robskillington. 

@benraskin92 yeah alpine is as small as busybox but has more features such package manager.

File: docker/Dockerfile-m3coordinator-dev (hunk 2 above)
Comment (nerd0) [reply docker/Dockerfile-m3coordinator-dev]: Sure fair point

**Pull request closed without merging**